solr/solrj/src/test/org/apache/solr/client/solrj/impl/ClusterStateProviderTest.java

```diff
@@ -47,19 +54,44 @@ public static void setupCluster() throws Exception {
         .configure();
   }
 
-  private ClusterStateProvider createClusterStateProvider() throws Exception {
-    return !usually() ? http2ClusterStateProvider() : zkClientClusterStateProvider();
+  @ParametersFactory
+  public static Iterable<String[]> parameters() throws NoSuchMethodException {
+    return List.of(
+        new String[] {"http2ClusterStateProvider"}, new String[] {"zkClientClusterStateProvider"});
   }
 
-  private ClusterStateProvider http2ClusterStateProvider() throws Exception {
-    return new Http2ClusterStateProvider(
-        List.of(cluster.getJettySolrRunner(0).getBaseUrl().toString()), null);
+  private static ClusterStateProvider http2ClusterStateProvider() {
+    try {
+      return new Http2ClusterStateProvider(
+          List.of(cluster.getJettySolrRunner(0).getBaseUrl().toString()), null);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
   }
 
-  private ClusterStateProvider zkClientClusterStateProvider() {
+  private static ClusterStateProvider zkClientClusterStateProvider() {
     return new ZkClientClusterStateProvider(cluster.getZkStateReader());
   }
 
+  private final Supplier<ClusterStateProvider> cspSupplier;
+
+  public ClusterStateProviderTest(String method) throws Exception {
```

Comment:
by using a String arg that refers to a method, the printed test execution is clear as to what is being invoked.  This uses reflection allows deferred execution of construction of the CSP, which is useful.